src/pages/UnlockedWallet/OverviewPage/AddressesContactsList.tsx

```diff
@@ -0,0 +1,217 @@
+import { Contact } from '@/types/contacts'
+import { currencies } from '@/utils/currencies'
+
+interface AddressesContactsListProps {
+  className?: string
+  limit?: number
+}
+
+const tabs = [
+  { value: 'addresses', label: i18next.t('Addresses') },
+  { value: 'contacts', label: i18next.t('Contacts') }
+]
+
+const AddressesContactsList = ({ className, limit }: AddressesContactsListProps) => {
+  const { t } = useTranslation()
+  const navigate = useNavigate()
+  const [isLoadingAddresses] = useAppSelector((s) => [s.addresses.loading])
+
+  const [currentTab, setCurrentTab] = useState<TabItem>(tabs[0])
+
+  return (
+    <Table isLoading={isLoadingAddresses} className={className}>
+      <TableTabBar
+        items={tabs}
+        onTabChange={(tab) => setCurrentTab(tab)}
+        activeTab={currentTab}
+        linkText={t('See more')}
+        onLinkClick={() => navigate('/wallet/addresses')}
+      />
+      {
+        {
+          addresses: <AddressesList limit={limit} />,
+          contacts: <ContactsList limit={limit} />
+        }[currentTab.value]
+      }
+    </Table>
+  )
+}
+
+const AddressesList = ({ className, limit }: AddressesContactsListProps) => {
+  const addresses = useAppSelector(selectAllAddresses)
+  const { data: price } = useGetPriceQuery(currencies.USD.ticker)
+
+  const [selectedAddress, setSelectedAddress] = useState<Address>()
+
+  const displayedAddresses = limit ? addresses.slice(0, limit) : addresses
+
+  return (
+    <motion.div {...fadeIn} className={className}>
+      {displayedAddresses.map((address) => (
+        <TableRow
+          key={address.hash}
+          role="row"
+          tabIndex={0}
+          onClick={() => setSelectedAddress(address)}
+          onKeyPress={() => setSelectedAddress(address)}
+        >
+          <Row>
+            <AddressColor>
+              {address.isDefault ? <Star color={address.color}>★</Star> : <DotIcon size={11} color={address.color} />}
+            </AddressColor>
+            {address.label ? (
+              <Column>
+                <Label>{address.label}</Label>
+                <Hash addressHash={address.hash} />
+              </Column>
+            ) : (
+              <Label>
+                <AddressEllipsed addressHash={address.hash} />
+              </Label>
+            )}
+            <TableCellAmount>
+              <Amount
+                value={calculateAmountWorth(BigInt(address.balance), price ?? 0)}
+                fadeDecimals
+                isFiat
+                suffix={currencies['USD'].symbol}
+                tabIndex={0}
+              />
+            </TableCellAmount>
+          </Row>
+        </TableRow>
+      ))}
+      <ModalPortal>
+        {selectedAddress && (
+          <AddressDetailsModal addressHash={selectedAddress.hash} onClose={() => setSelectedAddress(undefined)} />
+        )}
+      </ModalPortal>
+    </motion.div>
+  )
+}
+const ContactsList = ({ className, limit }: AddressesContactsListProps) => {
```

Comment:
FYI: as mentioned on Figma, we may end up removing this tab which clutters the screen while adding very little utility IMO. Using the wallet during the testing phase will help us decide if this is needed or not!